doc/api/cli.md

```diff
@@ -1969,6 +1969,9 @@ A regular expression that configures the test runner to only execute tests
 whose name matches the provided pattern. See the documentation on
 [filtering tests by name][] for more details.
 
+If both `--test-name-pattern` and `--test-skip-pattern` are supplied,
```

Comment:
This statement is used a few times throughout this PR, but it is a little vague. How about something along the lines of:

> If a test name matches both `--test-name-pattern` and `--test-skip-pattern`, the `--test-name-pattern` flag takes precedence.